build-system/tasks/runtime-test/helpers.js

```diff
@@ -249,9 +289,379 @@ function isLargeRefactor() {
   return filesChanged.length >= LARGE_REFACTOR_THRESHOLD;
 }
 
+function getBrowserConfig() {
+  const chromeFlags = [];
+  if (argv.chrome_flags) {
+    argv.chrome_flags.split(',').forEach(flag => {
+      chromeFlags.push('--'.concat(flag));
+    });
+  }
+
+  const options = new Map();
+  options
+    .set('chrome_canary', {browsers: ['ChromeCanary']})
+    .set('chrome_flags', {
+      browsers: ['Chrome_flags'],
+      customLaunchers: {
+        // eslint-disable-next-line
+        Chrome_flags: {
+          base: 'Chrome',
+          flags: chromeFlags,
+        },
+      },
+    })
+    .set('edge', {browsers: ['Edge']})
+    .set('firefox', {browsers: ['Firefox']})
+    .set('headless', {browsers: ['Chrome_no_extensions_headless']})
+    .set('ie', {
+      browsers: ['IE'],
+      customLaunchers: {
+        IeNoAddOns: {
+          base: 'IE',
+          flags: ['-extoff'],
+        },
+      },
+    })
+    .set('safari', {browsers: ['Safari']})
+    .set('saucelabs', {
+      browsers: [
+        'SL_Chrome',
+        'SL_Firefox',
+        'SL_Edge_17',
+        'SL_Safari_12',
+        'SL_IE_11',
+        // TODO(amp-infra): Evaluate and add more platforms here.
+        //'SL_Chrome_Android_7',
+        //'SL_iOS_11',
+        //'SL_iOS_12',
+        'SL_Chrome_Beta',
+        'SL_Firefox_Beta',
+      ],
+    })
+    .set('saucelabs_lite', {browsers: ['SL_Safari_12', 'SL_Firefox']});
```

Comment:
The `--saucelabs` / `--saucelabs_lite` separation was a hack to get `gulp test` to use different sets of browsers for unit / integraion tests. This can be simplified in a follow up PR to `--saucelabs`, which is treated differently by the two new targets.